build-scripts/eksd/build.sh

```diff
@@ -11,10 +11,15 @@ else
   (cd "${SNAPCRAFT_PART_BUILD}/eks-distro" && git fetch --all && git pull)
 fi
 
-EKS_TRACK=$(echo "${GIT_TAG}" | cut -f1,2 -d'.' | tr -s . - | cut -c2-)
+EKS_TRACK=$(echo "${KUBE_TRACK}" | cut -f1,2 -d'.' | tr -s . -)
 EKS_RELEASE=$(cat "${SNAPCRAFT_PART_BUILD}"/eks-distro/release/"${EKS_TRACK}"/production/RELEASE)
 
+snapcraftctl set-version "$KUBE_TRACK.$EKS_RELEASE"
```

Comment:
I don't think this is correct. For example, eks v1.23-5 release is running kubernetes version v1.23.9, https://github.com/aws/eks-distro/releases/tag/v1-23-eks-5

This would set the version to `1.23.5` instead, which is not correct. Is it not enough to instead change https://github.com/canonical/microk8s/pull/3430/files#diff-3e91c14d99827f2b9997c7495162c4e899f582f684827e688caa516849b906a2R12 to pull the kuberentes version from the eks branch instead of the upstream?

That way, we can set the kubernetes version to whatever the eks version needs, and no other changes would be needed.

Reply:
We already pull the kubernetes version from the manifest and set it accordingly in the build.
This change is mostly for the jenkins process, I would say that the snap version of the `eksd` set to `1-23-5` is a better fit since it follows a different versioning and release cycle and basically a fork. For example the current eksd version is `1-23-eks-5`, not `v1.23.9`.

I'm not sure if we should have the kubernetes version as the snap version. If we can add the eksd version as a seperate metadata and adapt the jenkins process accordingly, I would be fine with either. If not, it would create the need to maintain 2 seperate jenkins processes, because they would have some significant changes. We can apply the same jenkins jobs to `-eksd` with this change.